astropy/nddata/utils.py

```diff
@@ -42,11 +50,24 @@ def overlap_slices(large_array_shape, small_array_shape, position):
         ``small_array[slices_small]`` extracts the region that is inside the
         large array.
     """
+    if len(small_array_shape) != len(large_array_shape):
+        raise ValueError("Both arrays must have the same number of dimensions.")
+
+    if len(small_array_shape) != len(position):
+        raise ValueError("Position must have the same number of dimensions as array.")
+
     # Get edge coordinates
-    edges_min = [int(pos + 0.5 - small_shape / 2.) for (pos, small_shape) in
-                 zip(position, small_array_shape)]
-    edges_max = [int(pos + 0.5 + small_shape / 2.) for (pos, small_shape) in
-                 zip(position, small_array_shape)]
+    edges_min = [int(np.floor(pos + 0.5 - small_shape / 2.))
```

Comment:
why the addition of `np.floor`? For positive numbers it does the same as `int`, while I'd think that negative positions would need special treatment.